src/main/scala/com/microsoft/partnercatalyst/fortis/spark/streamfactories/RadioStreamFactory.scala

```diff
@@ -0,0 +1,96 @@
+package com.microsoft.partnercatalyst.fortis.spark.streamfactories
+
+import java.io.InputStream
+import java.net.URL
+import java.util.Locale
+
+import com.github.catalystcode.fortis.speechtotext.Transcriber
+import com.github.catalystcode.fortis.speechtotext.config.{OutputFormat, SpeechServiceConfig, SpeechType}
+import com.github.catalystcode.fortis.speechtotext.utils.Func
+import com.microsoft.partnercatalyst.fortis.spark.streamprovider.{ConnectorConfig, StreamFactory}
+import org.apache.spark.storage.StorageLevel
+import org.apache.spark.streaming.StreamingContext
+import org.apache.spark.streaming.dstream.{DStream, ReceiverInputDStream}
+import org.apache.spark.streaming.receiver.Receiver
+
+class RadioStreamFactory extends StreamFactory[RadioTranscription]{
+  override def createStream(ssc: StreamingContext): PartialFunction[ConnectorConfig, DStream[RadioTranscription]] = {
+    case ConnectorConfig("Radio", params) =>
+      val radioUrl = params("radioUrl")
+      val audioType = params("audioType")
+      val locale = params("locale")
+      val subscriptionKey = params("subscriptionKey")
+      val speechType = params("speechType")
+      val outputFormat = params("outputFormat")
+
+      new RadioInputDStream(
+        ssc, radioUrl, audioType, locale, subscriptionKey,
+        speechType, outputFormat, StorageLevel.MEMORY_ONLY)
+  }
+}
+
+class RadioInputDStream(
+  ssc: StreamingContext,
+  radioUrl: String,
+  audioType: String,
+  locale: String,
+  subscriptionKey: String,
+  speechType: String,
+  outputFormat: String,
+  storageLevel: StorageLevel
+) extends ReceiverInputDStream[RadioTranscription](ssc) {
+  override def getReceiver(): Receiver[RadioTranscription] = {
+    logDebug("Creating radio transcription receiver")
+    new TranscriptionReceiver(radioUrl, audioType, locale, subscriptionKey, speechType, outputFormat, storageLevel)
+  }
+}
+
+class TranscriptionReceiver(
+  radioUrl: String,
+  audioType: String,
+  locale: String,
+  subscriptionKey: String,
+  speechType: String,
+  outputFormat: String,
+  storageLevel: StorageLevel
+) extends Receiver[RadioTranscription](storageLevel) {
+
+  private val language = new Locale(locale).getLanguage
+  private var audioStream: InputStream = _
+  private var transcriber: Transcriber = _
+
+  private lazy val onTranscription = new Func[String] {
```

Comment:
`speechtotext.utils.Func` is part of the Java speech-to-text websocket package. Java doesn't have a native function interface so I had to make one.